fairlearn/adversarial/_adversarial_mitigation.py

```diff
@@ -0,0 +1,1114 @@
+        :math:`z` (sensitive features).
+
+    callbacks : callable
+        Callback function, called after every batch. For instance useable when
+        wanting to validate. Can also be a list of callback functions
+        Each callback function is passed two arguments :code:`self` (the
+        estimator instance) and :code:`step` (the completed iteration), and
+        may return a Boolean value. If the returned value is `True`, the
+        optimization algorithm terminates. This can be used to implement
+        *early stopping*.
+
+    cuda : str, default = None
+        A string to indicate which device to use when training. For instance,
+        set :code:`cuda='cuda:0'` to train on the first GPU. Only for PyTorch
+        backend.
+
+    warm_start : bool, default = False
+        Normally, when set to False, a call to fit triggers reinitialization,
+        which discards the models and intializes them again. Setting to
+        True triggers reuse of these models. Note: if pre-initialized models
+        were passed, the models (and their parameters) are never discarded.
+
+    random_state : int, RandomState instance, default = None
+        Controls the randomized aspects of this algorithm, such as shuffling.
+        Useful to get reproducible output across multiple function calls.
+
+    References
+    ----------
+    .. [1] Zhang, Lemoine, Mitchell `"Mitigating Unwanted Biases with
+       Adversarial Learning" <https://dl.acm.org/doi/pdf/10.1145/3278721.3278779>`_,
+       AIES, 2018.
+
+    """  # noqa : E501
+    # Hidden: predictor_loss, adversary_loss, y_transform,
+    # sf_transform
+
+    def __init__(self, **kwargs):
+        """Initialize model by setting the predictor loss and function."""
+        self._estimator_type = "classifier"
+        kwargs["y_transform"] = "classification"
+        kwargs["predictor_loss"] = "classification"
+        kwargs["adversary_loss"] = "auto"  # Removed choice for now.
+        kwargs["predictor_function"] = "classification"
+        kwargs["threshold_value"] = 0.5  # Removed choice for now.
+        super(AdversarialFairnessClassifier, self).__init__(**kwargs)
+
+
+class AdversarialFairnessRegressor(_AdversarialFairness, RegressorMixin):
+    r"""Train PyTorch or Tensorflow regressors while mitigating unfairness.
+
+    This estimator implements the supervised learning method proposed in
+    `"Mitigating Unwanted Biases with Adversarial Learning"
+    <https://dl.acm.org/doi/pdf/10.1145/3278721.3278779>`_ [1]_.
+    The training algorithm takes as input two neural network
+    models, a predictor model and an adversarial model, defined either as a
+    `PyTorch module
+    <https://pytorch.org/docs/stable/generated/torch.nn.Module.html>`_ or
+    `Tensorflow2 model
+    <https://www.tensorflow.org/api_docs/python/tf/keras/Model>`_. The API
+    follows conventions of `sklearn` estimators.
+
+    As per [1]_, the neural network models :code:`predictor_model` and
```

Comment:
Please adjust the description below similarly to the description I'm suggesting for the classifier above. The suggestions for the parameter descriptions below are exactly the same as for `AdversarialFairnessClassifier`.